forest/src/main.rs

```diff
@@ -6,23 +6,116 @@ mod daemon;
 mod logger;
 mod subcommand;
 
-use cli::{cli_error_and_die, Cli};
+use cli::{cli_error_and_die, Cli, DaemonConfig};
+
+use async_std::task;
+use daemonize_me::{Daemon, Group, User};
+use log::info;
+use raw_sync::events::{Event, EventInit};
+use raw_sync::Timeout;
+use shared_memory::ShmemConf;
 use structopt::StructOpt;
 
-#[async_std::main]
-async fn main() {
+use std::fs::File;
+use std::process;
+use std::time::Duration;
+
+const EVENT_TIMEOUT: Timeout = Timeout::Val(Duration::from_secs(20));
+
+// The parent process and the daemonized child communicate through an Event in
+// shared memory. The identity of the shared memory object is written to a local
+// file named .forest_daemon_ipc. The parent process is responsible for cleaning
+// up the local file and the shared memory object.
+fn ipc_shmem_conf() -> ShmemConf {
+    ShmemConf::new()
+        .size(Event::size_of(None))
+        .force_create_flink()
+        .flink(std::env::temp_dir().join("forest_daemon_ipc"))
```

Comment:
this defaults to `/tmp` on Unix, right? So running two detached instances of Forest (e.g. calibnet and mainnet) would fiasco, no? How about using [this nice crate](https://docs.rs/tempdir/latest/tempdir/), it's already used in some of the tests.

Reply:
Why not yes! Is there any guarantee that the temp dir is stable (so same after forking)?

Reply:
We can pick a random temporary file name before the fork.

Reply:
the temporary directory exists for the duration of the scope so probably it should be fine. Btw. my bad, I meant this crate https://docs.rs/tempfile/latest/tempfile/, it's more feature-rich.